ble/DiscoveredCharacteristic.h

```diff
@@ -83,6 +83,8 @@ class DiscoveredCharacteristic {
      */
     ble_error_t read(uint16_t offset = 0) const;
 
+    ble_error_t read(uint16_t offset, const GattClient::ReadCallback_t& onRead) const;
```

Comment:
I'm not sure this is the best API pattern. I accept that we need a way to setup callbacks for read, but it could also be `discoveredChar.read(offset).onRead(callback);`